src/__experimental__/components/date/date.component.js

```diff
@@ -191,15 +192,31 @@ class BaseDateInput extends React.Component {
       this.isOpening = false;
       return;
     }
+
     document.removeEventListener('click', this.closeDatePicker);
-    this.setState({ isDatePickerOpen: false }, () => {
+    this.setState({ isDatePickerOpen: false, shouldPickerOpen: false }, () => {
       this.isBlurBlocked = false;
       if (this.input !== document.activeElement) {
         this.handleBlur();
       }
     });
   };
 
+  handleClick = () => {
+    this.setState(_prevState => ({
+      shouldPickerOpen: !_prevState.shouldPickerOpen
+    }), () => {
+      if (!this.state.shouldPickerOpen) {
```

Comment:
just a thought.. I think it would be more readable if you re-aranged if/else statement here. You have check for Falsy in if and then drop into else. Maybe switch it around? Have a Thruthy IF?

Reply:
Good point. I'll re-write it so the IF statement is truthy and then the ELSE deals with the falsy. Thank you.